api/views/views.py

```diff
@@ -586,3 +592,51 @@ def post(self, request, format=None):
         except BaseException as e:
             logger.error(str(e))
             return HttpResponse(status=404)
+        
+class ZipListPhotosView_V2(APIView):
+    
+    def post(self, request):
+        import shutil
+
+        free_storage = shutil.disk_usage("/").free
+        data = dict(request.data)
+        if "image_hashes" not in data:      
+            return
+        photo_query=Photo.objects.filter(owner=self.request.user)
+        photos = photo_query.in_bulk(
+                data["image_hashes"]
+        )
+        if len(photos) == 0 :
+            return 
+        total_file_size = (
+            photo_query.aggregate(Sum("size"))["size__sum"]
+            or None
+        )
+        if free_storage < total_file_size:
+            return Response(data={'status': ""},status=507)
+        
+        filename=str(secrets.token_hex(nbytes=16)+'.zip')
+        user=self.request.user
+        job_id=create_download_job(LongRunningJob.JOB_DOWNLOAD_PHOTOS,user=user,photos=photos,filename=filename)
+        response={"job_id": job_id,'url':filename}
+
+        return Response(data=response,status=200)
+        
+   
+    def get(self,request):
+        try:
+            job=LongRunningJob.objects.get(job_id=request.data["job_id"])
+            if job.finished:
+                zip_file_path =  os.path.join("/protected_media/all_zip_folder/", request.data["url"])
```

Comment:
use constant from the settings (MEDIA_ROOT)

Reply:
done